libgnucash/engine/gnc-date.h

```diff
@@ -82,6 +82,14 @@ extern "C"
  * this stops working in 2038, we define our own:
  */
 typedef gint64 time64;
+/* A bit of a hack to create a type separate from the alias of int64_t so that
+ * compile-time dispatch can use the right KVP visitor.
+ */
+typedef struct
+{
+    time64 t;
+} Time64;
+
```

Comment:
The problem is that we can't differentiate between int64_t and time64 *even in C++*.  Aliases don't create new types, just new names for types, so the compiler doesn't see a difference between KvpValue<int64_t> and KvpValue<time64>.  We must use struct or class to create new types.

GValue doesn't care; in fact it depends on the sort of pointer games that you objected to in Transaction.c. But it's not feasible to separate the GValue type from the Kvp type because we pass GValues to Kvp.

Reply:
Ah, that helps. I had a vague recollection of strong typing in boost::mpl, but they do indeed use classes.